src/v/pandaproxy/json/requests/partitions.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "json/json.h"
+#include "model/fundamental.h"
+#include "pandaproxy/json/types.h"
+#include "seastarx.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <rapidjson/encodings.h>
+
+namespace pandaproxy::json {
+
+template<typename Encoding = rapidjson::UTF8<>>
+class partitions_request_handler {
+private:
+    enum class state {
+        empty = 0,
+        partitions,
+        topic_partition,
+        topic,
+        partition,
+    };
+
+    serialization_format _fmt = serialization_format::none;
+    state state = state::empty;
+
+public:
+    using Ch = typename Encoding::Ch;
```

Comment:
i thought all our alias as `snake_case` but this is a nit, no need to change.

Reply:
> i thought all our alias as `snake_case` but this is a nit, no need to change.

@senior7515 I was following the pattern from rapidjson - I'll fix this with the base parser handler.

Reply:
I already have some serialisation refactoring to do, so I'll leave it until then.